src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -1610,6 +1647,14 @@ static void parsing_action(grpc_exec_ctx *exec_ctx, void *arg, bool success) {
     ;
   if (i != t->read_buffer.count) {
     success = false;
+    gpr_slice_unref(t->optional_drop_message);
+    if (try_http_parsing(exec_ctx, t)) {
```

Comment:
This is probably the wrong thing to do unless it's the first slice received on a connection?